CHANGELOG.md

```diff
@@ -0,0 +1,24 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+
+This project adheres to [Semantic Versioning](http://semver.org/).
```

Comment:
I know this isn't legally binding text, but should this be stated with a warning? Something like "This project adheres to Semantic Versioning where possible." etc. Otherwise we might have people being surprised when a minor version change breaks existing functionality. We cannot really account for all cases like a library with a public API.